File: data/schema/career_qualitative.py
"""Schema for LLM Call #3: Qualitative Career Data & Pathway"""

from typing import List, Literal
from pydantic import BaseModel, Field


class WorkerTestimonial(BaseModel):
    quote: str
    sentiment: Literal['positive', 'neutral', 'negative']
    theme: str  # e.g., "work-life balance", "career growth"


class WorkExperience(BaseModel):
    typical_day: str
    work_environment: str
    challenges: List[str]
    rewards: List[str]
    testimonials: List[WorkerTestimonial]


class CurrentState(BaseModel):
    relevant_experience: List[str]
    skill_gaps: List[str]
    strengths: List[str]
    readiness_score: int = Field(ge=1, le=5)  # 1-5 scale


class PathwayStep(BaseModel):
    step: int
    title: str
    description: str
    timeframe: str  # e.g., "6 months", "1-2 years"
    priority: Literal['high', 'medium', 'low']
    resources: List[str]


class CareerPathway(BaseModel):
    user_current_state: CurrentState
    roadmap: List[PathwayStep]
    timeline: str  # Overall time estimate


class EducationResource(BaseModel):
    type: Literal['degree', 'bootcamp', 'online-course', 'certification']
    name: str
    provider: str
    duration: str
    cost: str  # e.g., "$5,000-10,000", "Free"
    relevance: int = Field(ge=1, le=5)  # 1-5 scale


class CertificationResource(BaseModel):
    name: str
    provider: str
    difficulty: Literal['beginner', 'intermediate', 'advanced']
    time_to_complete: str
    industry_recognition: int = Field(ge=1, le=5)  # 1-5 scale


class NetworkingResource(BaseModel):
    type: Literal['professional-org', 'conference', 'online-community']
    name: str
    description: str
    cost: str


class Resources(BaseModel):
    education: List[EducationResource]
    certifications: List[CertificationResource]
    networking: List[NetworkingResource]
    job_boards: List[str]


class AlternativeCareer(BaseModel):
    title: str
    onet_code: str
    similarity_reason: str
    transition_difficulty: Literal['easy', 'moderate', 'difficult']


class CareerQualitativeOutput(BaseModel):
    career_title: str
    onet_code: str
    summary: str  # LLM-generated comprehensive overview
    work_experience: WorkExperience
    career_pathway: CareerPathway
    resources: Resources
    alternative_careers: List[AlternativeCareer]

File: data/schema/career-qualitative.ts
// Schema for LLM Call #3: Qualitative Career Data & Pathway

export interface CareerQualitativeOutput {
  careerTitle: string;
  onetCode: string;
  summary: string; // LLM-generated comprehensive overview
  workExperience: {
    typicalDay: string;
    workEnvironment: string;
    challenges: string[];
    rewards: string[];
    testimonials: WorkerTestimonial[];
  };
  careerPathway: {
    userCurrentState: CurrentState;
    roadmap: PathwayStep[];
    timeline: string; // Overall time estimate
  };
  resources: {
    education: EducationResource[];
    certifications: CertificationResource[];
    networking: NetworkingResource[];
    jobBoards: string[];
  };
  alternativeCareers: AlternativeCareer[];
}

export interface WorkerTestimonial {
  quote: string;
  sentiment: 'positive' | 'neutral' | 'negative';
  theme: string; // e.g., "work-life balance", "career growth"
}

export interface CurrentState {
  relevantExperience: string[];
  skillGaps: string[];
  strengths: string[];
  readinessScore: number; // 1-5 scale
}

export interface PathwayStep {
  step: number;
  title: string;
  description: string;
  timeframe: string; // e.g., "6 months", "1-2 years"
  priority: 'high' | 'medium' | 'low';
  resources: string[];
}

export interface EducationResource {
  type: 'degree' | 'bootcamp' | 'online-course' | 'certification';
  name: string;
  provider: string;
  duration: string;
  cost: string; // e.g., "$5,000-10,000", "Free"
  relevance: number; // 1-5 scale
}

export interface CertificationResource {
  name: string;
  provider: string;
  difficulty: 'beginner' | 'intermediate' | 'advanced';
  timeToComplete: string;
  industry_recognition: number; // 1-5 scale
}

export interface NetworkingResource {
  type: 'professional-org' | 'conference' | 'online-community';
  name: string;
  description: string;
  cost: string;
}

export interface AlternativeCareer {
  title: string;
  onetCode: string;
  similarityReason: string;
  transitionDifficulty: 'easy' | 'moderate' | 'difficult';
}